btas/generic/cp_als.h

```diff
@@ -595,16 +647,13 @@ namespace btas {
           contract(1.0, ai, {1, 2}, ai, {1, 3}, 0.0, AtA[i], {2, 3});
         }
         is_converged = converge_test(A, AtA);
-      }
+      }while (count < max_als && !is_converged);
 
+      detail::get_fit(converge_test, epsilon);
+      epsilon = 1 - epsilon;
       // Checks loss function if required
-      if (calculate_epsilon) {
-        if (typeid(converge_test) == typeid(btas::FitCheck<Tensor>)) {
-          detail::get_fit(converge_test, epsilon);
-          epsilon = 1 - epsilon;
-        } else {
+      if (calculate_epsilon && typeid(converge_test) != typeid(btas::FitCheck<Tensor>)) {
```

Comment:
Only FitCheck gives info about the fit of the decomposition (\epsilon = 1- |T - \hat{T}| / |T| ) So if the user wants the actual epsilon and are using a different converge checking class it needs to be computed. But I will change the logic so that the `get_fit` function by default returns -1 and then this if doesn't have to use RTTI.